src/test/java/org/apache/maven/archiver/MavenArchiverTest.java

```diff
@@ -1444,54 +1447,76 @@ public void testParseOutputTimestamp()
-                1570300662000L );
+        assertThat( archiver.parseOutputTimestamp( "2019-10-05T18:37:42Z" ).getTime() )
+            .isEqualTo( 1570300662000L );
+        assertThat( archiver.parseOutputTimestamp( "2019-10-05T20:37:42+02:00" ).getTime() )
+            .isEqualTo( 1570300662000L );
+        assertThat( archiver.parseOutputTimestamp( "2019-10-05T16:37:42-02:00" ).getTime() )
+            .isEqualTo( 1570300662000L );
 
         // These must result in IAE because we expect extended ISO format only (ie with - separator for date and
         // : separator for timezone), hence the XXX SimpleDateFormat for tz offset
         // X SimpleDateFormat accepts timezone without separator while date has separator, which is a mix between
         // basic (no separators, both for date and timezone) and extended (separator for both)
-        try
-        {
-            archiver.parseOutputTimestamp( "2019-10-05T20:37:42+0200" );
-            fail();
-        }
-        catch ( IllegalArgumentException ignored )
-        {
-        }
-        try
-        {
-            archiver.parseOutputTimestamp( "2019-10-05T20:37:42-0200" );
-            fail();
-        }
-        catch ( IllegalArgumentException ignored )
-        {
-        }
+        assertThatExceptionOfType( IllegalArgumentException.class )
+            .isThrownBy( () -> archiver.parseOutputTimestamp( "2019-10-05T20:37:42+0200" ) );
+        assertThatExceptionOfType( IllegalArgumentException.class )
+            .isThrownBy( () -> archiver.parseOutputTimestamp( "2019-10-05T20:37:42-0200" ) );
+    }
 
-        // These unfortunately fail although the input is valid according to ISO 8601
-        // SDF does not allow strict telescoping parsing w/o permitting invalid input as depicted above.
-        // One has to use the new Java Time API for this.
-        try
-        {
-            archiver.parseOutputTimestamp( "2019-10-05T20:37:42+02" );
-            fail();
-        }
-        catch ( IllegalArgumentException ignored )
-        {
-        }
-        try
-        {
-            archiver.parseOutputTimestamp( "2019-10-05T20:37:42-02" );
-            fail();
-        }
-        catch ( IllegalArgumentException ignored )
-        {
-        }
+    @ParameterizedTest
+    @NullAndEmptySource
+    @ValueSource( strings = { ".", " ", "_", "-", "T", "/", "!", "!", "*", "ñ" } )
+    public void testEmptyParseOutputTimestampInstant( String value )
+    {
+        // Empty optional if null or 1 char
+        assertThat( MavenArchiver.parseBuildOutputTimestamp( value ) ).isEmpty();
+    }
+
+    @ParameterizedTest
+    @CsvSource( { "0,0", "1,1", "9,9", "1570300662,1570300662", "2147483648,2147483648",
+        "2019-10-05T18:37:42Z,1570300662", "2019-10-05T20:37:42+02:00,1570300662",
+        "2019-10-05T16:37:42-02:00,1570300662", "1988-02-22T15:23:47.76598Z,572541827",
+        "2011-12-03T10:15:30+01:00[Europe/Paris],1322903730",
```

Comment:
`2011-12-03T10:15:30+01:00[Europe/Paris]` is invalid. Should not pass.

Reply:
Sure, changed to OffsetDateTime to not accept the Zone.